common/network/neighbor.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __COMMON_NETWORK_NEIGHBOR_H__
+#define __COMMON_NETWORK_NEIGHBOR_H__
+
+#include "common/network/iota_packet.h"
+
+typedef enum connection_type_e { TCP, UDP } connection_type_t;
+
+typedef struct {
+  socket_addr_t endpoint;
+  int socket;
```

Comment:
some targets might not understand the notion of a socket?

could we place endpoint & socket and `connection_type` in some generic `void*` blob?

Reply:
The endpoint is the real host:port of the neighbor, should it be in the blob ?

Reply:
I think we only need some obscure identifier for the neighbour. That can be the memory address (i.e. `void*`) of the description struct.